app/models/concerns/filterable.rb

```diff
@@ -0,0 +1,13 @@
+module Filterable
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def filter(filtering_params)
+      results = where(nil)
+      filtering_params.each do |key, value|
+        results = results.public_send(key, value) if value.present?
```

Comment:
it's better to do a `results.where(key, value)` here. The guy from the blog post is not big on security :D